qiskit/visualization/matplotlib.py

```diff
@@ -548,34 +548,28 @@ def _get_coords(self, n_lines):
                             c_indxs.append(index)
                             break
 
-                # only add the gate to the anchors if it is going to be plotted.
-                if self._plot_barriers or not node.op._directive:
-                    for ii in q_indxs:
-                        self._q_anchors[ii].set_index(this_anc, layer_width)
-
                 # qubit coordinate
                 self._data[node]["q_xy"] = [
-                    self._q_anchors[ii].plot_coord(this_anc, layer_width, self._x_offset)
+                    self._q_anchors[ii].plot_coord(anc_x_index, layer_width, self._x_offset)
                     for ii in q_indxs
                 ]
                 # clbit coordinate
                 self._data[node]["c_xy"] = [
-                    self._c_anchors[ii].plot_coord(this_anc, layer_width, self._x_offset)
+                    self._c_anchors[ii].plot_coord(anc_x_index, layer_width, self._x_offset)
                     for ii in c_indxs
                 ]
                 # update index based on the value from plotting
-                this_anc = self._q_anchors[q_indxs[0]].gate_anchor
+                anc_x_index = self._q_anchors[q_indxs[0]].get_x_index()
                 self._data[node]["c_indxs"] = c_indxs
 
             # adjust the column if there have been barriers encountered, but not plotted
             barrier_offset = 0
             if not self._plot_barriers:
                 # only adjust if everything in the layer wasn't plotted
                 barrier_offset = -1 if all(nd.op._directive for nd in layer) else 0
-            prev_anc = this_anc + layer_width + barrier_offset - 1
+            prev_x_index = anc_x_index + layer_width + barrier_offset - 1
 
-        anchors = [self._q_anchors[ii].get_index() for ii in self._qubits_dict]
-        return max(anchors) if anchors else 0
+        return prev_x_index + 1
```

Comment:
Logically this seems like the maximum will always be the last index, but just to check: is there any scenario in which `layer_width` or `barrier_offset` will cause a `prev_x_index` to be smaller than its predecessor?

Reply:
In this line,
```
570    prev_x_index = anc_x_index + layer_width + barrier_offset - 1
```
`layer_width >= 1` and `barrier_offset >= -1` and `anc_x_index` was incremented, so the lowest `prev_x_index` would be is unchanged from last time.

Reply:
Yeah, that's what I'd read out of the code, I just wanted to check that there was nothing more in the underlying meaning of the variables that meant that logically it could be decremented.